crates/pathfinder/src/p2p_network/sync_handlers/v1/tests.rs

```diff
@@ -0,0 +1,554 @@
+
+#[rstest]
+#[case(0, 1, Direction::Forward, Some(1))]
+#[case(0, I64_MAX, Direction::Forward, Some(I64_MAX))]
+#[case(1, I64_MAX, Direction::Forward, None)]
+#[case(0, 1, Direction::Backward, None)]
+#[case(1, 1, Direction::Backward, Some(0))]
+#[case(I64_MAX, 1, Direction::Backward, Some(I64_MAX - 1))]
+#[case(I64_MAX, I64_MAX, Direction::Backward, Some(0))]
+#[case(I64_MAX, I64_MAX + 1, Direction::Backward, None)]
+#[test]
+fn get_next_block_number(
+    #[case] start: u64,
+    #[case] step: u64,
+    #[case] direction: Direction,
+    #[case] expected: Option<u64>,
+) {
+    assert_eq!(
+        super::get_next_block_number(
+            BlockNumber::new_or_panic(start),
+            Step::from(Some(step)),
+            direction
+        ),
+        expected.map(BlockNumber::new_or_panic)
+    );
+}
+
+mod empty_reply {
+    use super::I64_MAX;
+    use crate::p2p_network::sync_handlers::v1::{
+        get_bodies, get_events, get_headers, get_receipts, get_transactions,
+    };
+    use assert_matches::assert_matches;
+    use fake::{Fake, Faker};
+    use p2p_proto_v1::block::{BlockBodiesRequest, BlockHeadersRequest};
+    use p2p_proto_v1::common::Iteration;
+    use p2p_proto_v1::event::EventsRequest;
+    use p2p_proto_v1::receipt::ReceiptsRequest;
+    use p2p_proto_v1::transaction::TransactionsRequest;
+    use pathfinder_storage::Storage;
+    use rand::Rng;
+    use rstest::rstest;
+    use tokio::sync::mpsc;
+    use tokio::sync::mpsc::error::TryRecvError;
+
+    fn zero_limit() -> Iteration {
+        Iteration {
+            limit: 0,
+            ..Faker.fake()
+        }
+    }
+
+    fn invalid_start() -> Iteration {
+        Iteration {
+            start_block: rand::thread_rng().gen_range(I64_MAX + 1..=u64::MAX),
+            ..Faker.fake()
+        }
+    }
+
+    macro_rules! define_test {
+        ($name:ident, $uut_name:ident, $request:tt) => {
+            #[rstest]
+            #[case(zero_limit())]
+            #[case(invalid_start())]
+            #[tokio::test]
+            async fn $name(#[case] iteration: Iteration) {
+                let storage = Storage::in_memory().unwrap();
+                let (tx, mut rx) = mpsc::channel(1);
+                // Clone the sender to make sure that the channel is not prematurely closed
+                $uut_name(&storage, $request { iteration }, tx.clone())
+                    .await
+                    .unwrap();
+                // No reply should be sent
+                assert_matches!(rx.try_recv().unwrap_err(), TryRecvError::Empty);
+            }
+        };
+    }
+
+    define_test!(headers, get_headers, BlockHeadersRequest);
+    define_test!(bodies, get_bodies, BlockBodiesRequest);
+    define_test!(transactions, get_transactions, TransactionsRequest);
+    define_test!(receipts, get_receipts, ReceiptsRequest);
+    define_test!(events, get_events, EventsRequest);
+}
+
+/// Property tests, grouped to be immediately visible when executed
+mod prop {
```

Comment:
one test per handler, each test tries exploring the input space using a compound strategy that allows almost all kinds of values but slightly favors inputs that produce some overlapping range with what is generated in the fake storage